### PR TITLE
feat(ModalSelectorLayout)| support disabled items

### PR DESCRIPTION
**Pull request by aryelu**
### What changed
added support for disabled items in ModalSelectorLayout
...

### Why it changed
needed to support for disabled items in ModalSelectorLayout
...

---

- [x] Change is tested
- [x] Change is documented
- [x] Build is green
- [x] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

File: src/ModalSelectorLayout/ModalSelectorLayout.js [hunk 1]
```diff
@@ -148,7 +149,8 @@ export default class ModalSelectorLayout extends WixComponent {
     searchValue: '',
     selectedItems: [],
     shouldShowNoResultsFoundState: false,
-    isEmpty: false
+    isEmpty: false,
+    disabledCount: 0
```
Comment (argshook) [inline src/ModalSelectorLayout/ModalSelectorLayout.js]: is this not used?

File: src/Selector/Selector.js [hunk 2]
```diff
@@ -62,7 +64,7 @@ class Selector extends WixComponent {
         onClick={this._onClick(id)}
         >
         { toggleType === 'checkbox' ?
-          <Checkbox dataHook="toggle" checked={isSelected}/> :
+          <Checkbox dataHook="toggle" checked={isSelected} disabled={isDisabled}/> :
```
Comment (argshook) [inline src/Selector/Selector.js]: i suppose same should go for radio button as well

File: src/ModalSelectorLayout/ModalSelectorLayout.js [hunk 3]
```diff
@@ -305,6 +309,7 @@ export default class ModalSelectorLayout extends WixComponent {
       .then(({items: itemsFromNextPage, totalCount}) => {
         if (this.state.searchValue === searchValue) { // react only to the resolve of the relevant search
           const newItems = [...items, ...itemsFromNextPage];
+          const newDisabledCount = newItems.filter(item => item.disabled).length;
```
Comment (argshook) [inline src/ModalSelectorLayout/ModalSelectorLayout.js]: this is used in `setState` but where is that number used from state? did i miss something?

File: src/ModalSelectorLayout/ModalSelectorLayout.js [hunk 4]
```diff
@@ -247,11 +249,13 @@ export default class ModalSelectorLayout extends WixComponent {
     );
   }
 
+  _isDisabled = item => this.props.multiple && item.disabled;
```
Comment (argshook) [inline src/ModalSelectorLayout/ModalSelectorLayout.js]: does `this.props.multiple` really need to be here?

File: src/ModalSelectorLayout/ModalSelectorLayout.js [hunk 5]
```diff
@@ -247,11 +249,13 @@ export default class ModalSelectorLayout extends WixComponent {
...
+
   _renderItems() {
     const {items, selectedItems} = this.state;
     const {imageSize, imageShape, multiple} = this.props;
 
-    const isSelected = item => !!selectedItems.find(({id}) => item.id === id);
+    const isSelected = item => !!selectedItems.find(({id}) => item.id === id) || this._isDisabled(item);
```
Comment (argshook) [inline src/ModalSelectorLayout/ModalSelectorLayout.js]: maybe
`const isSelected = item => !item.disabled && !!selectedItems.find(({id}) => item.id === id)`

File: src/ModalSelectorLayout/ModalSelectorLayout.js (hunk 1 above)
Comment (aryelu) [reply src/ModalSelectorLayout/ModalSelectorLayout.js]: I update it in https://github.com/wix/wix-style-react/pull/1550/files/d96c32871f364f48bc868c8089700488d95f7f5d#diff-c262611a545d555687acab8089e548e3R323

File: src/ModalSelectorLayout/ModalSelectorLayout.js (hunk 4 above)
Comment (aryelu) [reply src/ModalSelectorLayout/ModalSelectorLayout.js]: I did this only from `multiple` mode, I don't know if this specific implementation (disable and set as selected) can work with radio button

File: src/ModalSelectorLayout/ModalSelectorLayout.js (hunk 5 above)
Comment (aryelu) [reply src/ModalSelectorLayout/ModalSelectorLayout.js]: I did it only for `multiple` mode

File: src/ModalSelectorLayout/ModalSelectorLayout.js (hunk 3 above)
Comment (aryelu) [reply src/ModalSelectorLayout/ModalSelectorLayout.js]: I just override the number, we don't have access to `items` in the construction time

**Pull request merged**